### PR TITLE
TST: Relax tolerance in upfirdn test_modes

### PR DESCRIPTION
**Pull request by grlee77**
This PR relaxes the tolerance used for single precision cases for a test case introduced recently in #10552.

Some test cases at scipy-wheels are failing with the current setting of 1e-6. For example:
https://travis-ci.org/MacPython/scipy-wheels/jobs/604686365#L15033

When I test locally on 64-bit linux with Numpy 1.17.3 the arrays being compared are actually identical and would pass for any tolerance. I am not sure where the source of difference is for the cases on scipy-wheels CI, but think 1e-5 should be acceptable for single-precision comparisons.

### CODE REVIEW

File: scipy/signal/tests/test_upfirdn.py [hunk 1]
```diff
@@ -228,4 +228,8 @@ def test_modes(self, size, h_len, mode, dtype):
         ypad = upfirdn(h, xpad, up=1, down=1, mode='constant')
         y_expected = ypad[npad:-npad]
 
-        assert_allclose(y, y_expected, atol=1e-6, rtol=1e-6)
+        if dtype in (np.float32, np.complex64):
+            atol = rtol = 1e-5
+        else:
+            atol = rtol = 1e-12
+        assert_allclose(y, y_expected, atol=atol, rtol=rtol)
```
Comment (larsoner) [inline scipy/signal/tests/test_upfirdn.py]: Would something like `atol = rtol = np.finfo(dtype).eps * 1e2` work?
```
>>> np.finfo(np.complex64).eps * 1e2
1.1920928955078125e-05
>>> np.finfo(np.complex128).eps * 1e2
2.220446049250313e-14
```
It cuts down on lines and seems more principled, but if it doesn't work then what you have seems fine

Comment (ilayn) [reply scipy/signal/tests/test_upfirdn.py]: I meant the differences. The values are in the same order of magnitude as far as I can see. `rtol` is espcially helpful when the compared quantities have order of magnitude drastically changing.

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge assuming CIs are happy (ignoring `dual_annealing` failures fixed by #10993)

File: scipy/signal/tests/test_upfirdn.py (hunk 1 above)
Comment (grlee77) [reply scipy/signal/tests/test_upfirdn.py]: Okay.  I am hesitant to make it tighter than it currently is as I cannot reproduce the failures seen on scipy-wheels when testing locally (even `atol = rtol = 0` passes on my system).

**Comment by grlee77** [comment]
Okay, looks like all CI passed except for the unrelated error Eric mentioned in the prior comment.

**Review by larsoner — APPROVED** [review]

**Comment by larsoner** [comment]
@ilayn feel free to merge if you can live with the `atol/rtol` choices

**Comment by larsoner** [comment]
I'll go ahead and merge since the atol/rtol at least tends to follow conventions in that file, but @ilayn if you want we can make a follow-up PR to clean some of them up if need be

**Pull request merged**